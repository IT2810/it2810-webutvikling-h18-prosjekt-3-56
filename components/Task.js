import React, { Component } from 'react';
import { Text, View, StyleSheet} from 'react-native';
import Swipeout from 'react-native-swipeout';


export default class Task extends Component {

  constructor(props){
    super(props);
    this.swipeoutBtns = [
      {
        text:'Done',
        backgroundColor: "#2ecc71",
        //onPress = function to change state, task has been completed
        onPress: () => this.props.click(props.id)

      }
    ]
  }

  render() {
    return (
      <Swipeout autoClose = {true} style = {styles.margin} right = {this.swipeoutBtns}>
        <View style = {styles.container}>
          <Text style = {[styles.fineText, styles.font]}>{this.props.text}</Text>
        </View>
      </Swipeout>
    );
  }
}


const styles = StyleSheet.create({
  container:{
    height: 'auto',
    backgroundColor:'white'
  },
  font:{
    fontSize: 25,
    padding:3
  },
  margin:{
    marginBottom:1
  }
})
