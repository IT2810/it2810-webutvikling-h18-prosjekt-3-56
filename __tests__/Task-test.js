import React from 'react';
import Task from '../components/Task.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Task text = "Test everything"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
