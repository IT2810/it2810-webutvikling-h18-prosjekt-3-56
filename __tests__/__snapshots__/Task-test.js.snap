// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dbddde",
        "overflow": "hidden",
      },
      Object {
        "marginBottom": 1,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {},
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "height": "auto",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Object {
              "fontSize": 25,
              "padding": 3,
            },
          ]
        }
      >
        Test everything
      </Text>
    </View>
  </View>
  <View />
  <View />
</View>
`;
